.github/workflows/wheels.yml

```diff
@@ -174,8 +174,8 @@ jobs:
           CIBW_CONFIG_SETTINGS_WINDOWS: "setup-args=--vsenv"
           CIBW_REPAIR_WHEEL_COMMAND_WINDOWS: bash build_tools/github/repair_windows_wheels.sh {wheel} {dest_dir}
           CIBW_BEFORE_TEST_WINDOWS: bash build_tools/github/build_minimal_windows_image.sh ${{ matrix.python }}
-          CIBW_TEST_REQUIRES: pytest
           CIBW_BEFORE_TEST: bash {project}/build_tools/wheels/cibw_before_test.sh
+          CIBW_TEST_REQUIRES: pytest pandas
```

Comment:
I am not sure why we need this, since the dev version of pandas is already installed in `cibw_before_test.sh`.

I checked [the logs](https://github.com/scikit-learn/scikit-learn/actions/runs/10113263237/job/27969234807?pr=29572#step:6:1884) and this do not cause `pip` to override the dev wheel with the stable release wheel:

```
...
      + pip install pytest pandas
  Collecting pytest
    Downloading pytest-8.3.2-py3-none-any.whl.metadata (7.5 kB)
  Requirement already satisfied: pandas in /tmp/tmp.ZWPQ7oIjQL/venv/lib/python3.13/site-packages (3.0.0.dev0+1252.gd6c9941b73)
...
```

Reply:
`CIBW_TEST_REQUIRES` is used by all the other builds though (i.e. not free-threaded ones).

I guess doing it this way makes it easier to update when eventually when numpy, scipy and pandas have a release with a free-threaded wheel: remove `CIBW_BEFORE_TEST` and `build_tools/wheels/cibw_before_test.sh`.